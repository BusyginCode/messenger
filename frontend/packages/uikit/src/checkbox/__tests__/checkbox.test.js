import React from 'react'
import { render } from 'enzyme'
import Checkbox from '../checkbox'

describe('uikit/checkbox', () => {
  it('should renders correctly', () => {
    const wrapper = render(<Checkbox value="some-val" checked label="label" />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should renders correctly label="label"', () => {
    const wrapper = render(<Checkbox value="some-val" checked />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should renders correctly label=<span />', () => {
    const wrapper = render(<Checkbox value="some-val" checked label={<span />} />)
    expect(wrapper).toMatchSnapshot()
  })
})
