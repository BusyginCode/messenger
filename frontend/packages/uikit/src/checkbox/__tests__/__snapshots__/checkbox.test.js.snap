// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uikit/checkbox should renders correctly 1`] = `
<label
  class="box align_center"
>
  <span
    class="dt-checkbox dt-checkbox-checked"
  >
    <input
      checked=""
      class="dt-checkbox-input"
      data-role="checkbox_item"
      type="checkbox"
    />
    <span
      class="dt-checkbox-inner"
    />
  </span>
  <p
    class="box base general s left"
    data-role="checkbox_text_label"
    style="margin-left:8px"
  >
    label
  </p>
</label>
`;

exports[`uikit/checkbox should renders correctly label="label" 1`] = `
<span
  class="dt-checkbox dt-checkbox-checked"
>
  <input
    checked=""
    class="dt-checkbox-input"
    data-role="checkbox_item"
    type="checkbox"
  />
  <span
    class="dt-checkbox-inner"
  />
</span>
`;

exports[`uikit/checkbox should renders correctly label=<span /> 1`] = `
<label
  class="box align_center"
>
  <span
    class="dt-checkbox dt-checkbox-checked"
  >
    <input
      checked=""
      class="dt-checkbox-input"
      data-role="checkbox_item"
      type="checkbox"
    />
    <span
      class="dt-checkbox-inner"
    />
  </span>
  <span />
</label>
`;
